File: src/main/java/com/syfe/personalfinance/dto/AuthDto.java
package com.syfe.personalfinance.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

public class AuthDto {

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class RegisterRequest {
        @NotBlank(message = "Username (email) is required")
        @Email(message = "Username must be a valid email address")
        private String username;

        @NotBlank(message = "Password is required")
        private String password;

        @NotBlank(message = "Full name is required")
        private String fullName;

        private String phoneNumber;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class LoginRequest {
        @NotBlank(message = "Username (email) is required")
        @Email(message = "Username must be a valid email address")
        private String username;

        @NotBlank(message = "Password is required")
        private String password;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class UserResponse {
        private Long id;
        private String username;
        private String fullName;
        private String phoneNumber;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class RegisterResponse {
        private String message;
        private Long userId;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class SimpleMessageResponse {
        private String message;
    }
}

File: src/main/java/com/syfe/personalfinance/dto/AuthDto.kt
package com.syfe.personalfinance.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

class AuthDto {

    data class RegisterRequest(
        @field:NotBlank(message = "Username (email) is required")
        @field:Email(message = "Username must be a valid email address")
        val username: String = "",

        @field:NotBlank(message = "Password is required")
        val password: String = "",

        @field:NotBlank(message = "Full name is required")
        val fullName: String = "",

        val phoneNumber: String? = null
    )

    data class LoginRequest(
        @field:NotBlank(message = "Username (email) is required")
        @field:Email(message = "Username must be a valid email address")
        val username: String = "",

        @field:NotBlank(message = "Password is required")
        val password: String = ""
    )

    data class UserResponse(
        val id: Long? = null,
        val username: String = "",
        val fullName: String? = null,
        val phoneNumber: String? = null
    )

    data class RegisterResponse(
        val message: String = "",
        val userId: Long? = null
    )

    data class SimpleMessageResponse(
        val message: String = ""
    )
}
